### ISSUE TITLE
Mitigating Veracode Scan Issues

### ISSUE THREAD

**Issue by PRISMAY** [jspreadsheet/ce#1099]
Hi Paul,
Our company uses Veracode code scanning to find issues in code that may be vulnerable to attacks.  There are a few hits on the jexcel files and wanted to share this with you and see if there's a way this can be fixed.  The jexcel version we're using right now is 3.8.2.

The code that's triggering the issues (link below).  From jexcel.js

```
if (eval('typeof(' + tokens[i] + ') == "undefined"')) {
...
var res = eval(evalstring + expression.substr(1));
```
From jsuites.js

`eval(script[i].innerHTML);`

http://cwe.mitre.org/data/definitions/95.html

Thank you and let me know if you have questions.

**Comment by pphod** [comment]
Thank you for your message.

There were included many new options thinking on the level of security you would like to bring to your applications, and particular from v3 to v4 there are a big differences on that regards. For example you can disable formulas, so eval would never be used. You can secureFormulas which will only execute (excel like formulas). And you can use stripHTML which only accept innerText on DOM elements.

Those are some of the variables introducted on the most recent versions.

            // Execute formulas
            parseFormulas:true,
            autoIncrement:true,
            autoCasting:true,
            secureFormulas:true,
            stripHTML:true,
            stripHTMLOnCopy:false,


In regards to jSuites. The eval mentioned is no longer on the jexcel distributions. So, it won't be included on jSuites.js or jSuites.basic.js, althoght we still using for a very special case in a special distribution not related to jexcel.

That means, I would recommend you update Jsuites to v3 and Jexcel to v4 to get the updates which we have mitigated a lot of potential issues, giving the developer the power to decide the level of security.

Best regards,

**Comment by PRISMAY** [comment]
Thanks Paul.

**Comment by PRISMAY** [comment]
I grabbed jExcel via npm and it downloaded the 4.3.0 version.  I noticed the jsuites distribution included all these files:
```
jsuites.basic.js
jsuites.js
jsuites.layout.js
jsuites.mobile.js
jsuites.webcomponents.js
```
Do we need all the files or just `jsuites.js` and `jsuites.basic.js` per what you mention above?

The jExcel dist files include `jexcel.webcomponent.js` which I assume is needed for formulas when enabled.

Thanks for your help.
